frontend/src/framework/InitialSettings.ts

```diff
@@ -58,17 +58,20 @@ export class InitialSettings {
     isSet(settingName: string, type: keyof InitialSettingsSupportedTypes): boolean {
         return this.get(settingName, type) !== undefined;
     }
+}
 
-    applyToStateOnMount(
-        settingName: string,
-        type: keyof InitialSettingsSupportedTypes,
-        stateSetter: (value: any) => void
-    ): void {
-        React.useEffect(() => {
-            const setting = this.get(settingName, type);
+export function applyToStateOnMount(
```

Comment:
This function needs a new name since it has lost its naming context by no longer being a member of `InitialSettings` and it doesn't do the application on mount anymore. 
Maybe something along the lines of `maybeApplyInitialSettingsToState()`, `applyInitialSettingsToState()`, `applyNewInitialSettingsToState()` ...

Reply:
Going with `applyInitialSettingsToState` as `InitialSettings` are only applied once in the lifetime of a module.